cli/command/manifest/push.go

```diff
@@ -42,32 +47,63 @@ type pushRequest struct {
 	insecure      bool
 }
 
+type yamlManifestList struct {
+	Image     string
+	Manifests []yamlManifest
+}
+
+type yamlManifest struct {
+	Image    string
+	Platform manifestlist.PlatformSpec
+}
+
 func newPushListCommand(dockerCli command.Cli) *cobra.Command {
 	opts := pushOpts{}
 
 	cmd := &cobra.Command{
-		Use:   "push [OPTIONS] MANIFEST_LIST",
+		Use:   "push [OPTIONS] [MANIFEST_LIST]",
 		Short: "Push a manifest list to a repository",
-		Args:  cli.ExactArgs(1),
+		Args:  cli.RequiresRangeArgs(0, 1),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			opts.target = args[0]
+			if len(args) == 1 {
+				opts.target = args[0]
+			}
 			return runPush(dockerCli, opts)
 		},
 	}
 
 	flags := cmd.Flags()
-	flags.BoolVarP(&opts.purge, "purge", "p", false, "Remove the local manifest list after push")
-	flags.BoolVar(&opts.insecure, "insecure", false, "Allow push to an insecure registry")
+	flags.BoolVarP(&opts.purge, "purge", "p", false, "remove the local manifest list after push")
+	flags.BoolVar(&opts.insecure, "insecure", false, "allow push to an insecure registry")
+	flags.StringVar(&opts.file, "filename", "", "create, annotate, and push using a yaml file")
```

Comment:
I wonder if `file` should be a boolean flag that changes the positional argument from a "local manifest target" to "a filepath".  I'm not sure if that's more or less expected than having a conditional argument.

I think it would at least make the validation easier, because with a bool flag a positional argument would always be necessary.